packages/block-library/src/image/edit.js

```diff
@@ -108,11 +108,39 @@ export function ImageEdit( {
 		align,
 		metadata,
 	} = attributes;
+
 	const [ temporaryURL, setTemporaryURL ] = useState( attributes.blob );
 
 	const [ contentResizeListener, { width: containerWidth } ] =
 		useResizeObserver();
 
+	const [ placeholderResizeListener, { width: placeholderWidth } ] =
+		useResizeObserver();
+
+	// Parse width of the image block attribute.
+	const parseWidth = ( widthValue ) => {
+		const parsed = parseInt( widthValue, 10 );
+		return isNaN( parsed ) ? null : parsed;
+	};
+
+	const parsedWidth = parseWidth( width );
+	const parsedPlaceholderWidth = placeholderWidth || null;
+
+	// Get the effective width of the image placeholder use the minimum of placeholder width
+	// or image attribute width if both are defined. If only one is defined, use that one.
+
+	const getEffectiveWidth = () => {
+		if ( parsedPlaceholderWidth !== null && parsedWidth !== null ) {
+			return Math.min( parsedPlaceholderWidth, parsedWidth );
+		}
+		return parsedPlaceholderWidth !== null
+			? parsedPlaceholderWidth
+			: parsedWidth;
+	};
+
+	const effectiveWidth = getEffectiveWidth();
+	const isLargeContainer = effectiveWidth !== null && effectiveWidth > 160;
```

Comment:
Rather than adding a new `useResizeObserver` hook, can't we reuse the one that already exists (`containerWidth`)? Also, the width/height obtained by the `useResizeObserver()` hook should only be [a number type or null type](https://github.com/WordPress/gutenberg/blob/18164f0fcd09a282157f02bba7fd737012cc44fd/packages/compose/src/hooks/use-resize-observer/index.tsx#L16-L19). Taking this into consideration, I expect the logic here can be simplified further.

Reply:
Thanks, @t-hamano, for the review. I tried using the existing containerWidth, but since it was applied outside the <figure>, it always returned the content width. When using an image inside the grid, it didn't provide the exact width of the image block.
I'll update the effective width logic to account for these return types.

Reply:
Ah, I misunderstood that the existing hook was for getting the width of the block itself. I think adding a new hook makes sense 👍